apps/modernization-ui/src/apps/page-builder/components/TableMenu/TableMenu.tsx

```diff
@@ -34,6 +34,12 @@ export const TableMenu = ({ tableType, searchQuery, setSearchQuery, onDownloadIc
         }
     }, [searchQuery]);
 
+    useEffect(() => {
+        if (!keywords || keywords === '') {
+            setSearchQuery('');
+        }
+    }, [keywords]);
```

Comment:
question: Should this case be handled in the on-click?  it's changing search query without having to hit submit, not sure if that is desired behavior or not.

Reply:
it's setting the query, but it won't search until 'Search' is clicke.
It searches for '' when the user clicks 'X' to clear the searchQuery, search is run automatically only then.